drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1235,6 +1334,15 @@ static void endrx_isr(const struct device *dev)
 	 */
 	const int rx_amount = nrf_uarte_rx_amount_get(uarte) + async_rx->flush_cnt;
 
+#ifdef CONFIG_HAS_NORDIC_DMM
```

Comment:
as above - this could should compile fine for legacy targets as well - `dmm_buffer_in_release` is mocked for them